app/src/main/java/com/murrayc/galaxyzoo/app/SubjectFragment.java

```diff
@@ -427,7 +438,63 @@ private void updateShareActionIntent() {
         //shareIntent.putExtra(Intent.EXTRA_STREAM, mUriImageStandard);
         //shareIntent.setType("image/*");
 
+        if (mUriStandardRemote!=null) {
+            GetImageBitmapAsyncTask getImageBitmapAsyncTask = new GetImageBitmapAsyncTask(){
+                @Override
+                protected void onPostExecute(Uri uri) {
+                    shareIntent.setType("image/*");
+                    shareIntent.putExtra(Intent.EXTRA_STREAM, uri);
+                    shareIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                    mShareActionProvider.setShareIntent(shareIntent);
+                }
+            };
+            getImageBitmapAsyncTask.execute(mUriStandardRemote);
+        }
+
 
         mShareActionProvider.setShareIntent(shareIntent);
```

Comment:
> I would still leave the instantiation outside onPostExecute, since that leaves less main thread work to be done 
> after the AsyncTask is completed.

OK. That sounds reasonable, but it would be nice to have a comment explaining that choice.